cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -337,7 +337,10 @@ public ApplicationWithPrograms updateApp(ApplicationId appId, AppRequest appRequ
     // ownerAdmin.getImpersonationPrincipal will give the owner which will be impersonated for the application
     // irrespective of the version
     if (!Objects.equals(ownerAdmin.getImpersonationPrincipal(appId), appRequest.getOwnerPrincipal())) {
-      throw new ConflictException(String.format("Updating %s is not supported.", Constants.Security.PRINCIPAL));
+      throw new ConflictException(String.format("The specified %s: '%s' is not same as existing one. " +
```

Comment:
Although thinking over it app re-deployment is an app update in our case. I will remove the already exists part from ApplicationVerificationStage.